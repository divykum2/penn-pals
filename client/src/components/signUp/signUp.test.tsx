 /**
* @jest-environment jsdom
*/
import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from 'react-router-dom';
import SignUp from './signUp';
import LoginPage from '../loginPage/loginPage';

test('renders page', async() => {
  render(<Router><SignUp /></Router>);

  expect(screen.getByText('Sign Up')).toBeInTheDocument();
  expect(screen.getByText('Username')).toBeInTheDocument();
  expect(screen.getByText('First Name')).toBeInTheDocument();
  expect(screen.getByText('Last Name')).toBeInTheDocument();
  expect(screen.getByText('Email Address')).toBeInTheDocument();
  expect(screen.getByText('Password')).toBeInTheDocument();
});

test('testing navigation', () => {
    render(<Router><SignUp /></Router>);

  const button = screen.getByRole('button', {name: "Sign Up"});
  fireEvent.click(button);
  render(<Router><LoginPage/></Router>)
  expect(screen.getByText("Sign in")).toBeInTheDocument()
  
})

test('renders app logo', () => {
    render(<Router><SignUp/></Router>);
    const image = screen.getByAltText('logo');
    expect(image).toHaveAttribute('src', 'logo.png');
})

test('Activity feed matches snapshot', () => {
    const component = renderer.create(<Router><SignUp/></Router>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});