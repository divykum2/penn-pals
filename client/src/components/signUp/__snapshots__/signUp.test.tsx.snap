// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity feed matches snapshot 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs css-nzinn5-MuiContainer-root"
>
  <div
    className="MuiBox-root css-binzgt"
  >
    <img
      alt="logo"
      height="60"
      src="logo.png"
    />
    <div
      className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1capexe-MuiAvatar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="LockOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zM9 6c0-1.66 1.34-3 3-3s3 1.34 3 3v2H9V6zm9 14H6V10h12v10zm-6-3c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2z"
        />
      </svg>
    </div>
    <h1
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      Sign up
    </h1>
    <form
      className="MuiBox-root css-178yklu"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="firstName"
              id="firstName-label"
            >
              First Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="given-name"
                autoFocus={true}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="firstName"
                name="firstName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    First Name
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-1b3l6lk-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="lastName"
              id="lastName-label"
            >
              Last Name
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="family-name"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="lastName"
                name="lastName"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Last Name
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="email"
              id="email-label"
            >
              Email Address
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="email"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="email"
                name="email"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Email Address
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="username"
              id="username-label"
            >
              Username
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="username"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="username"
                name="username"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="text"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Username
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-required MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink={false}
              htmlFor="password"
              id="password-label"
            >
              Password
              <span
                aria-hidden={true}
                className="MuiFormLabel-asterisk MuiInputLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
              >
                 
                *
              </span>
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoComplete="new-password"
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                disabled={false}
                id="password"
                name="password"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={true}
                type="password"
                value=""
              />
              <fieldset
                aria-hidden={true}
                className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="css-yjsfm1"
                >
                  <span>
                    Password
                     
                    *
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1bou8hu-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text-aline="center"
          type="button"
        >
          Sign Up
        </button>
      </div>
      <div
        className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
      >
        <div
          className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            Already have an account? Sign in
          </a>
        </div>
      </div>
    </form>
  </div>
</main>
`;
